### ISSUE TITLE
USER ISSUE: Can't sell or buy some items w/o skill learned

### ISSUE THREAD

**Issue by Scheigy** [StrangeLoopGames/EcoIssues#5779]
**Version**: 0.7.2.3 beta

**Steps to Reproduce**:
In a store, try to buy or sell stuff that need a skill to be learned by scroll. The item won't appear in the search list if you haven't unlock the specific skill.

**Expected behavior**:
We should be able to buy and sell any item no matter which skill we learned. It's currently preventing the reselling of object for players who can't craft this object.

**Actual behavior**:
Example: Can't buy/sell Yeast before using the Advanced Cooking Skill Scroll. Same behavior with Elk Wellington for Advanced Bakery also.

**Comment by Matikari** [comment]
I noticed this as well, As a chef I wanted to buy raw meat, but it was not listed for me until after someone took the hunting skill.

**Comment by Scheigy** [comment]
Matikari is right. When anyone on the server learn the skill, everyone can then buy or sell it in their shop.

I guess it was an intended behavior that missed its indication the changelog.

**Comment by MarjieAVolk** [comment]
The items should become available when the required skill is researched.  No one should need to learn the specialty to unlock it.  Is that not the case?

**Comment by Scheigy** [comment]
Sorry for the late answer. My tests were biased because in my local server, I used the /give command for the scrolls and so I never did any research using the table. This mean the OnUsed function for scrolls should be overriden to call whatever function trigger the behavior you described.

Here are the steps I made for the last test:

- I created a new world. In the store, I can't find the Stone Road in the search filter.
- I research the Stone Construction and use the scroll.
- After a disconnect (this may be linked to another issue), I'm able to research the Basic Engineering 
- I research the Basic Engineering. As soon as the research is done and without needing to collect the book, the Stone Road is now appearing in the search filter area.

So it's working as you describe it should be working. I probably wasn't aware of that mechanic in our first server because I discovered items existed as long as we discovered them.  But imo this mechanic is still confusing and might need some reworking.

For example, you are able to find Iron Ingots in the list, without researching Smelting first, but not Copper Ingots (same tree).

My conclusion would be this is a really low priority issue, and is more of a suggestion than a real issue.
